sklearnex/tests/_utils.py

```diff
@@ -112,7 +112,7 @@ def __getitem__(self, key):
 )
 
 
-def gen_models_info(algorithms):
+def gen_models_info(algorithms, required_inputs=["X", "y"]):
```

Comment:
this is a not straightforward function and given that we are refactoring it would be a good opportunity to improve variable names (specifically i) - would recommend making i,j,k slightly more informative

Reply:
I have made the iterables names more informative and added documentation to the functions to explain their use.